src/UnoFlipController.java

```diff
@@ -53,29 +51,50 @@ public void actionPerformed(ActionEvent e) {
                     null,
                     null);
 
+            // numPlayers used to create the amount of players the user has selected
             int numPlayers = 0;
-            if (result == JOptionPane.OK_OPTION) {
+            if (result == JOptionPane.OK_OPTION)
+            {
+                // Parse the input given by the user from the drop-down to use to create players
```

Comment:
instead of breaking down each line, try to write comments that explain a larger block of code - a high level explanation. In this case, saying something like "get the number of players to create the game for from user input" is enough because you aren't doing anything wonky. Comments like "Tell model the number of players in the game" are redundant, because anyone can tell that is what you are doing from the code itself (this.model.setNumPlayers(numPlayers);)